auth-web/src/components/auth/staff/continuation-application/HomeJurisdictionInformation.vue

```diff
@@ -231,122 +231,144 @@
+    async function downloadAffidavitDocument (): Promise<void> {
+      await download(state.continuationIn?.foreignJurisdiction?.affidavitFileKey,
+        state.continuationIn?.foreignJurisdiction?.affidavitFileName)
     }
 
-    if (country === JurisdictionLocation.US) {
-      const state = UsaJurisdiction.find(usa => usa.value === region)?.text
-      return (state ? `${state}, US` : 'USA')
+    /** Downloads the specified authorization document. */
+    async function downloadAuthorizationDocument (item: { fileKey: string, fileName: string }): Promise<void> {
+      await download(item.fileKey, item.fileName)
     }
 
-    return IntlJurisdictions.find(intl => intl.value === country)?.text || null
-  }
-
-  get identifier (): any {
-    return this.continuationIn?.foreignJurisdiction?.identifier
-  }
-
-  get legalName (): any {
-    return this.continuationIn?.foreignJurisdiction?.legalName
-  }
-
-  get taxId (): any {
-    return this.continuationIn?.foreignJurisdiction?.taxId
-  }
-
-  get incorporationDate (): string {
-    const date = DateUtils.yyyyMmDdToDate(this.continuationIn?.foreignJurisdiction?.incorporationDate)
-    return DateUtils.dateToPacificDate(date, true)
-  }
-
-  get authorizationFiles (): any {
-    return this.continuationIn?.authorization?.files
-  }
-
-  get authorizationDate (): string {
-    const date = DateUtils.yyyyMmDdToDate(this.continuationIn?.authorization?.date)
-    return DateUtils.dateToPacificDate(date, true)
-  }
-
-  /**
-   * Whether a continuation in director affidavit is required.
-   * Is true if the business is a Continued In ULC from Alberta or Nova Scotia.
-   */
-  get isContinuationInAffidavitRequired (): boolean {
-    const entityType = this.continuationIn?.nameRequest?.legalType
-    const country = this.continuationIn?.foreignJurisdiction?.country
-    const region = this.continuationIn?.foreignJurisdiction?.region
-
-    return (
-      entityType === CorpTypes.ULC_CONTINUE_IN &&
-      country === 'CA' &&
-      (region === 'AB' || region === 'NS')
-    )
-  }
-
-  /** Downloads the director affidavit document. */
-  async downloadAffidavitDocument (): Promise<void> {
-    await this.download(this.continuationIn?.foreignJurisdiction?.affidavitFileKey,
-      this.continuationIn?.foreignJurisdiction?.affidavitFileName)
-  }
+    /** Downloads the specified document. */
+    async function download (documentKey: string, documentName: string): Promise<void> {
+      if (!documentKey || !documentName) return // safety check
+
+      state.isDownloading = true
+      await BusinessService.downloadDocument(documentKey, documentName).catch(error => {
+        // eslint-disable-next-line no-console
+        console.log('downloadDocument() error =', error)
+        state.dialogTitle = 'Unable to download document'
+        state.dialogText = 'An error occurred while downloading the document. Please try again.'
+        const v = errorDialogComponent.value as any; v.open()
```

Comment:
Travis, I did this to circumvent the error "property 'open' does not exist on type 'unknown'".

I assume/hope that this will work out in Vue 3. What do you think?

Reply:
will be fine for Vue3, just typing